File: IT/Oggetti/Java/Person3.java
/* file Person3.java */

class Person {
	private String name;
	private String surname;

	public Person(String name, String surname) {
		setName(name);
		setSurname(surname);
	}

	/* setter e getter del nome */

	public void setName(String name) {
		this.name = !name.isEmpty() ? name : "John";
	}

	public String getName() {
		return this.name;
	}

	/* setter e getter del cognome */

	public void setSurname(String surname) {
		this.surname = !surname.isEmpty() ? surname : "Doe";
	}

	public String getSurname() {
		return this.surname;
	}

	/* dichiarazione del metodo walk() */

	public void walk() {
		System.out.println(this.getName() + " " + this.getSurname() + " passeggia."); 
	}
}

class Env {
	public static void main(String [] args) {
		Person p = new Person("Walt", "Disney");
		p.walk();

		// nome e cognome vuoti non sono validi: vengono usati valori di default
		p = new Person("", "");
		p.walk();
	}
}

File: IT/Oggetti/C++/Person3.cpp
/* file Person3.cpp */

#include <iostream>

using std::cout;
using std::endl;
using std::string;

class Person {
	private:
		string name;
		string surname;

	public:
		Person(string, string);
		void setName(string name);
		void setSurname(string surname);
		string getName();
		string getSurname();

		/* dichiarazione della funzione membro walk() */

		void walk();
};

Person::Person(string name, string surname)
{
	this->setName(name);
	this->setSurname(surname);
}

/* implementazione dei setter e getter del nome */

void Person::setName(string name) {
	this->name = !name.empty() ? name : "John";
}

string Person::getName() 
{
	return this->name;
}

/* implementazione dei setter e getter del cognome */ 

void Person::setSurname(string surname) {
	this->surname = !surname.empty() ? surname : "Doe";
}

string Person::getSurname()
{
	return this->surname;
}

/* implementazione della funzione membro walk() */

void Person::walk()
{
	cout << this->getName() << " " << this->getSurname() << " passeggia." << endl;
}

int main()
{
	Person p("Walt", "Disney");
	p.walk();

	p = Person("", "");
	p.walk();
}

